src/librustc_mir/build/expr/into.rs

```diff
@@ -187,8 +192,15 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                             // we have to do it; this overwrites any `break`-assigned value but it's
                             // always `()` anyway
                             this.cfg.push_assign_unit(exit_block, source_info, destination);
+
+                            out_terminator = TerminatorKind::Goto { target: loop_block };
```

Comment:
it would be nice if we indicated what `out_terminator` means in terms of the diagram above; I find this a bit hard to decipher (kind of a pre-existing problem :)